docs/USAGE.md

```diff
@@ -0,0 +1,296 @@
+    Operates on the value of a StreamBlock. Combines all child blocks of type `block_name` as 
+    children of a single ListBLock which is a child of the parent StreamBlock.
+- `operations.StreamChildrenToStreamBlockOperation`  
+    Operates on the value of a StreamBlock. Note that `block_names` here is a list of block types 
+    and not a single block type unlike `block_name` in the previous operation. Combines each child
+    block of a type in `block_names` as children of a single StreamBlock which is a child of the 
+    parent StreamBlock. 
+- `operations.StreamChildrenToStructBlockOperation`  
+    Moves each StreamBlock child of the given type inside a new StructBlock
+
+    A new StructBlock will be created as a child of the parent StreamBlock for each child block of
+    the given type, and then that child block will be moved from the parent StreamBlocks children
+    inside the new StructBlock as a child of that StructBlock.
+    
+    For example, consider the following StreamField definition,
+
+    ```
+    mystream = StreamField([("char1", CharBlock()) ...], ...)
+    ```
+
+    Then the stream data would look like the following,
+
+    ```
+    [
+        { "type": "char1", "value": "Value1", ... },
+        { "type": "char1", "value": "Value2", ... },
+        ...
+    ]
+    ```
+
+    And if we define the operation like this,
+
+    ```
+    StreamChildrenToStructBlockOperation("char1", "struct1")
+    ```
+
+    Our altered stream data would look like this,
+
+    ```
+    [
+        ...
+        { "type": "struct1", "value": { "char1": "Value1" } },
+        { "type": "struct1", "value": { "char1": "Value2" } },
+        ...
+    ]
+    ```
+
+    **NOTE:** Block ids are not preserved here since the new blocks are structurally different than 
+    the previous blocks.
+
+
+## Other Operations
+
+- `operations.AlterBlockValueOperation`
+
+# Making Custom Operations
+
+## Basic Usage
+
+While this package comes with a set of operations for common use cases, there may be many
+instances where you need to define your own operation for mapping data. Making a custom operation
+is fairly straightforward. All you need to do is extend the `BaseBlockOperation` class and 
+define the `apply` method. 
+
+For example, if we want to do something with the value of all matching blocks,
+
+```python
+from wagtail_streamfield_migration_toolkit.operations import BaseBlockOperation
+
+class MyBlockOperation(BaseBlockOperation):
+    def __init__(self, name):
+        super().__init__()
+        self.name = name
+        
+    def apply(self, block_value):
+        # do something with the block value
+        new_block_value = do_something(block_value)
+        return new_block_value
+
+```
+
+## block_value
+
+Note that depending on the type of block we're dealing with, the `block_value` which is passed to
+`apply` may take different structures.
+
+For non structural blocks, the value of the block will be passed directly.
```

Comment:
A small example would be useful here (e.g. `(for example, in the case of a CharBlock, the value will be the string value)`)